examples/samv71_xplained_ultra/pwm/main.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+
+using namespace Board;
+
+// Complementary output on PWM0 Channel2 High / Low outputs
+using OutH2 = GpioC19;
+using OutL2 = GpioD26;
+
+int main()
+{
+	Board::initialize();
+
+	MODM_LOG_INFO << "PWM Test" << modm::endl;
+
+	Pwm0::connect<OutH2::PwmH2, OutL2::PwmL2>();
+
+	Pwm0::initialize();
+	// Period 1500 => MCLK / 1500 = 100 kHz
+	Pwm0::setPeriod(Pwm0::Channel::Ch2, 1500);
+	// 500/1500 = 33.3% duty-cycle
+	Pwm0::setDutyCycle(Pwm0::Channel::Ch2, 500);
+
+	constexpr auto mode = Pwm0::ChannelMode()
+		.setClock(Pwm0::ChannelClock::Mck)
+		.setDeadTimeGeneration(Pwm0::DeadTimeGeneration::Enabled);
```

Comment:
Meh. Very cool, but kinda a one-off inconstent API. Isn't `Pwm0::ChannelClock::Mck | Pwm0::DeadTimeGeneration::Enabled` possible?

Reply:
Actually, that is not really the most useful thing. A lot of the options like `Pwm0::ChannelClock` are not bit flags. If you have an enum with values corresponding to constants (like 0, 1, 2, 3, 4) bit or-ing them is certainly not what you want to do. Users could easily specify values twice and get an invalid value as a result.

The only way to handle that case would be with `modm::Configuration` and that is actually not the cleanest API in my opinion. If you want to define a config variable it can't be `constexpr` and `const` because you need to call the mutating `Configuration::set()` function to set each option.

I can change it to use that but from a user perspective the API is not going to be better.

Reply:
Ok, then lets leave it like this and deal with this later. Perhaps there are better ways to do the register API in C++20 now.